editor/editor_asset_installer.cpp

```diff
@@ -82,229 +81,259 @@ void EditorAssetInstaller::open(const String &p_path, int p_depth) {
-			} else {
-				ti->set_tooltip_text(0, res_path);
-			}
-
-			ti->set_metadata(0, res_path);
+			ti = _create_file_item(parent_item, path, &num_file_conflicts);
 		}
-
-		status_map[E] = ti;
+		file_item_map[E] = ti;
 	}
 
+	asset_title_label->set_text(asset_name);
 	if (num_file_conflicts >= 1) {
-		asset_contents->set_text(vformat(TTR("Contents of asset \"%s\" - %d file(s) conflict with your project:"), asset_name, num_file_conflicts));
+		asset_conflicts_label->set_text(vformat(TTRN("%d file conflicts with your project", "%d files conflict with your project", num_file_conflicts), num_file_conflicts));
+		asset_conflicts_label->add_theme_color_override("font_color", get_theme_color(SNAME("error_color"), EditorStringName(Editor)));
 	} else {
-		asset_contents->set_text(vformat(TTR("Contents of asset \"%s\" - No files conflict with your project:"), asset_name));
+		asset_conflicts_label->set_text(TTR("No files conflict with your project"));
+		asset_conflicts_label->remove_theme_color_override("font_color");
 	}
 
 	popup_centered_ratio(0.5);
 	updating = false;
 }
 
+TreeItem *EditorAssetInstaller::_create_dir_item(TreeItem *p_parent, const String &p_path, HashMap<String, TreeItem *> &p_item_map) {
+	TreeItem *ti = tree->create_item(p_parent);
+	ti->set_cell_mode(0, TreeItem::CELL_MODE_CHECK);
+	ti->set_checked(0, true);
+	ti->set_editable(0, true);
+
+	ti->set_text(0, p_path.get_file() + "/");
+	ti->set_icon(0, get_theme_icon(SNAME("folder"), SNAME("FileDialog")));
+	ti->set_metadata(0, String());
+
+	p_item_map[p_path] = ti;
+	return ti;
+}
+
+TreeItem *EditorAssetInstaller::_create_file_item(TreeItem *p_parent, const String &p_path, int *r_conflicts) {
+	TreeItem *ti = tree->create_item(p_parent);
+	ti->set_cell_mode(0, TreeItem::CELL_MODE_CHECK);
+	ti->set_checked(0, true);
+	ti->set_editable(0, true);
+
+	String file = p_path.get_file();
+	String extension = file.get_extension().to_lower();
+	if (extension_icon_map.has(extension)) {
+		ti->set_icon(0, extension_icon_map[extension]);
+	} else {
+		ti->set_icon(0, generic_extension_icon);
+	}
+	ti->set_text(0, file);
+
+	String res_path = "res://" + p_path;
+	if (FileAccess::exists(res_path)) {
+		*r_conflicts += 1;
+		ti->set_custom_color(0, get_theme_color(SNAME("error_color"), EditorStringName(Editor)));
+		ti->set_tooltip_text(0, vformat(TTR("%s (already exists)"), res_path));
+		ti->set_checked(0, false);
+		ti->propagate_check(0);
+	} else {
+		ti->set_tooltip_text(0, res_path);
+	}
+
+	ti->set_metadata(0, res_path);
+
+	return ti;
+}
+
+void EditorAssetInstaller::_check_has_toplevel() {
+	// Check if the file structure has a distinct top-level directory. This is typical
+	// for archives generated by GitHub, etc, but not for manually created ZIPs.
+
+	toplevel_prefix = "";
+	skip_toplevel_check->set_pressed(false);
+	skip_toplevel_check->set_disabled(true);
+	skip_toplevel_check->set_tooltip_text(TTR("This asset doesn't have one root directory, so it can't be ignored."));
```

Comment:
```suggestion
	skip_toplevel_check->set_tooltip_text(TTR("This asset doesn't have a root directory, so it can't be ignored."));
```

Reply:
I tried to underline that it's looking for the one single root folder, but if it's still unclear (or clear even with your suggestion), then I can make the change.